explorer/testdata/linked_list/linked_list.carbon

```diff
@@ -0,0 +1,96 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
```

Comment:
I think we should also have some more focused unit tests that exercise the features of `Optional` in isolation, because those will be easier to debug.

Reply:
ok

Reply:
Done!